argparse_manpage/manpage.py

```diff
@@ -205,26 +222,114 @@ def __str__(self):
+            extra_description = description_section.content
+        lines.extend(self.mf.format_parser(self.parser, extra_description=extra_description))
 
-        if self.parser.epilog != None:
+        comments_section = self.get_extra_section("comments")
+        if self.parser.epilog or comments_section:
             lines.append("")
             lines.append('.SH COMMENTS')
-            lines.append(self.format_text(self.parser.epilog))
+            if comments_section:
+                lines.append(comments_section.content)
+            else:
+                lines.append(self.format_text(self.parser.epilog))
 
-        # Additional Section
-        for section in self.parser._manpage:
-            lines.append('.SH {}'.format(section['heading'].upper()))
-            lines.append(self.format_text(section['content']))
+        # Additional sections
+        for section in self.parser._manpage: # pylint: disable=protected-member
+            if section["heading"] not in SPECIAL_MANPAGE_SECTIONS:
+                lines.append('.SH {}'.format(section['heading'].upper()))
+                lines.append(section['content'])
 
         lines.append("")
         lines.extend(self.mf.format_footer(self._data))
-        return "\n".join(lines).strip("\n") + "\n"
+
+        # Finally add --include sections that match text in the page
+        final_lines = []
+        for line in lines:
+            final_lines.append(line)
+            for match in self._match_texts:
+                if match['match_text'] in line:
+                    final_lines.append(match['content'])
+
+        return "\n".join(final_lines).strip("\n") + "\n"
+
+    def get_extra_section(self, heading):
+        """
+        Return supplementary section for the `Manpage` (created with
+        `--include`), or `None`
+        """
+        for section in self.parser._manpage: # pylint: disable=protected-member
+            if section["heading"] == heading:
+                return section
+        return None
+
+    def add_section(self, heading, position, content):
+        """
+        Add a supplementary section to a `Manpage`
+        """
+        # Sections that need special treatment
+        heading = heading.lower()
+        if heading in ("author", "distribution"):
+            if heading == "author":
+                self._data['authors'] = [content]
+            elif heading == "distribution":
+                self._data['url'] = content
+        section = self.get_extra_section(heading)
+        if section is None:
+            section = {"heading": heading, "content": ""}
+            self.parser._manpage.append(section) # pylint: disable=protected-member
+        if position == '<':
+            section["content"] = content + section["content"]
+        elif position == '=':
+            section["content"] = content
+        elif position == '>':
+            section["content"] += content
+        else:
+            raise ValueError(f"invalid position {position}")
```

Comment:
Can you please avoid using the f-string patterns?  We try to ship argparse-manpage for RHEL 7, and there's (by default) Python 2.7 (reason for the EL7 failure).

Reply:
I've fixed this.